### PR TITLE
Blocking subscribe methods for convenience

### PR DESCRIPTION
**Pull request by akarnokd**
Mirroring some `Observable.subscribe` methods on `BlockingObservable`.

Related discussion: #3063.

### CODE REVIEW

File: src/main/java/rx/observables/BlockingObservable.java [hunk 1]
```diff
@@ -477,4 +473,227 @@ private void awaitForComplete(CountDownLatch latch, Subscription subscription) {
             throw new RuntimeException("Interrupted while waiting for subscription to complete.", e);
         }
     }
+    
+    /**
+     * Runs the source observable to a terminal event, ignoring any values and rethrowing any exception.
+     */
+    @Experimental
+    public void run() {
```
Comment (abersnaze) [inline src/main/java/rx/observables/BlockingObservable.java]: can this be renamed to `subscribe()`?

Comment (akarnokd) [inline src/main/java/rx/observables/BlockingObservable.java]: There is already a method named `subscribe()`. I've put in this alias for discoverability.

File: src/main/java/rx/observables/BlockingObservable.java [hunk 2]
```diff
@@ -477,4 +473,227 @@ private void awaitForComplete(CountDownLatch latch, Subscription subscription) {
...
+        run();
+    }
+    
+    /**
+     * Subscribes to the source and calls the given action on the current thread and rethrows any exception wrapped
+     * into OnErrorNotImplementedException.
+     * @param onNext the callback action for each source value
+     */
+    @Experimental
+    public void subscribe(final Action1<? super T> onNext) {
```
Comment (abersnaze) [inline src/main/java/rx/observables/BlockingObservable.java]: should this be an alias for `forEach(Action1)`?

Comment (akarnokd) [inline src/main/java/rx/observables/BlockingObservable.java]: They have different behavior. `forEach` executes the action on the thread that emits the `onNext` events and awaits the terminal event in the current thread. This method calls the action on the current thread.

Comment (abersnaze) [inline src/main/java/rx/observables/BlockingObservable.java]: Could you add a comment to the javadoc for both methods highlighting the difference between them?

**Comment by akarnokd** [comment]
Done.

**Pull request merged**